framework/log/Dispatcher.php

```diff
@@ -80,7 +80,7 @@ public function __construct($config = [])
     {
         // ensure logger gets set before any other config option
         if (isset($config['logger'])) {
-            $this->setLogger($config['logger']);
+            $this->setLogger(is_array($config['logger']) ? Yii::createObject($config['logger']) : $config['logger']);
```

Comment:
I think that could be a BC break as we allowed the logger to be any class before this. `Instance::ensure` will fail if someone passes a custom implementation.